packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -232,6 +232,38 @@ describe('CDK Include', () => {
     }, ResourcePart.CompleteDefinition);
   });
 
+  test('correctly parses Conditions and the Condition resource attribute', () => {
+    const cfnTemplate = includeTestTemplate(stack, 'resource-attribute-condition.json');
+    const alwaysFalseCondition = cfnTemplate.getCondition('AlwaysFalseCond');
+    const cfnBucket = cfnTemplate.getResource('Bucket');
+
+    expect(cfnBucket.cfnOptions.condition).toBe(alwaysFalseCondition);
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('resource-attribute-condition.json'),
+    );
+  });
+
+  test('reflects changes to a retrieved CfnCondition object in the resulting template', () => {
+    const cfnTemplate = includeTestTemplate(stack, 'resource-attribute-condition.json');
+    const alwaysFalseCondition = cfnTemplate.getCondition('AlwaysFalseCond');
+
+    alwaysFalseCondition.expression = core.Fn.conditionEquals(1, 2);
+
+    expect(stack).toMatchTemplate({
+      "Conditions": {
+        "AlwaysFalseCond": {
+          "Fn::Equals": [1, 2],
+        },
+      },
+      "Resources": {
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Condition": "AlwaysFalseCond",
+        },
+      },
+    });
+  });
+
```

Comment:
Any other operation that a customer must be able to do once a `CfnCondition` is retrieved from a template, besides use it on an L1 and re-assign the `expression`? 

Anything to verify that the [`expression`](https://github.com/aws/aws-cdk/blob/986e2814e072f0334d8470a2d60ea73dcceadfe4/packages/%40aws-cdk/core/lib/cfn-condition.ts#L22) is correctly parsed out?

Reply:
Not really, it's not easy to check the correctness of the condition expression (and anyway, if we did that validation, it probably shouldn't be done only for include - it should be done for all uses of condition).